flowey/flowey_lib_common/src/run_cargo_build.rs

```diff
@@ -297,7 +313,25 @@ fn rename_output(
     cargo_out_dir: &Path,
     dry_run: bool,
 ) -> Result<CargoBuildOutput, anyhow::Error> {
-    let do_rename = |ext: &str, no_dash: bool| {
+    fn rename_or_copy(from: impl AsRef<Path>, to: impl AsRef<Path>) -> std::io::Result<()> {
+        let res = fs_err::rename(from.as_ref(), to.as_ref());
+
+        let needs_copy = match res {
```

Comment:
Is there any concern that items that fall into `needs_copy` will be left behind by some cleanup tasks?

Reply:
The paths we are talking about here are conceptually "owned" by either flowey itself (and thus, cleaned up by flowey / adjacent infra), or `cargo`, which has its own cleanup mechanism (namely - `cargo clean`).

So we should be in the clear here

Reply:
That said - if folks think that this fallback-to-copy approach is reasonable, it might be useful to add an extra `fs_err::remove_file` step, just in case.